service/rpc/availability.go

```diff
@@ -0,0 +1,68 @@
+package rpc
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+
+	"github.com/celestiaorg/celestia-node/service/share"
+)
+
+const heightAvailabilityEndpoint = "/data_available"
+
+// AvailabilityResponse represents the response to a
+// `/data_available` request.
+type AvailabilityResponse struct {
+	Available   bool   `json:"available"`
+	Probability string `json:"probability_of_availability"`
+}
+
+func (h *Handler) handleHeightAvailabilityRequest(w http.ResponseWriter, r *http.Request) {
+	heightStr := mux.Vars(r)[heightKey]
+	height, err := strconv.Atoi(heightStr)
+	if err != nil {
+		writeError(w, http.StatusBadRequest, heightAvailabilityEndpoint, err)
+		return
+	}
+
+	header, err := h.header.GetByHeight(r.Context(), uint64(height))
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, heightAvailabilityEndpoint, err)
+		return
+	}
+
+	availResp := &AvailabilityResponse{
+		Probability: strconv.FormatFloat(
+			share.ProbabilityOfAvailability(share.DefaultSampleAmount), 'g', -1, 64),
```

Comment:
This is incorrect for the FullAvailability case. Meaning it always takes value from the number of samples being sampled which only applies to LightAvailability. It seems like the problem can be solved by extending the Availability interface explained in the comment above so that each impl makes the probability calculation inside. For the full availability, we would need to calculate the number of shares by doing EDSSize^2 and putting that into the formula.

Reply:
Don't we just return 1 for full availability given if the square can be reconstructed, the probability that it is available (to that full node) is 100%

Reply:
1 is fine

Reply:
Hm, we could do this w/ metrics though.

Reply:
Metrics are a good addition to this, yes, but mainly the probability needs to be exposed to a user who wants to know it. Thinking further, we might need to add a runtime feature to change the number of samples a light node does sample, and then the probability n will be stored in the cache availability.

Reply:
Is there anything you'd like to add in this current PR or can we make an issue with all the ideas documented above? @Wondertan

Reply:
I would make 2 issues out of what we have here:
* Report probability even for failed availability calls. E.g light client was able to find only N samples our of M or full node wasn't able to reconstruct, but was able to find N shares. From the gotten shares, we can get probability.
* Ability to modify the amount of sampled shares in the runtime for the Light Node.

Reply:
https://github.com/celestiaorg/celestia-node/issues/876

Reply:
and https://github.com/celestiaorg/celestia-node/issues/877

Reply:
Cool thanks, can i close this thread?

Reply:
Yep

Reply:
Resolving